src/com/palmergames/bukkit/towny/event/damage/TownBlockExplosionTestEvent.java

```diff
@@ -0,0 +1,62 @@
+package com.palmergames.bukkit.towny.event.damage;
+
+import com.palmergames.bukkit.towny.object.Town;
+import com.palmergames.bukkit.towny.object.TownBlock;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * An event thrown when Towny will determine the PVP status of 
+ * a townblock, or plot, in a town.
+ */
+public class TownBlockExplosionTestEvent extends Event {
+
+		private static final HandlerList handlers = new HandlerList();
+		private final TownBlock townBlock;
+		private boolean explosion;
+
+		public TownBlockExplosionTestEvent(TownBlock townBlock, boolean explosion) {
+			this.townBlock = townBlock;
+			this.setExplosion(explosion);
+		}
+
+
+		public static HandlerList getHandlerList() {
+			return handlers;
+		}
+
+		public HandlerList getHandlers() {
+			return handlers;
+		}
+
+		/**
+		 * @return the TownBlock which is having its explosion status decided.
+		 */
+		public TownBlock getTownBlock() {
+			return townBlock;
+		}
+
+		/**
+		 * @return the Town where this test is made, should never return null.
+		 */
+		@Nullable
+		public Town getTown() {
+			return townBlock.getTownOrNull();
+		}
```

Comment:
I feel like this should be a NotNull since we expect it to never be null.

Ideally, pass the Town instance through the event contstructor and return that so that IJ doesn't warn about getTownOrNull possibly returning null.